auto_tests/toxav_mt_test.cc

```diff
@@ -0,0 +1,498 @@
+    while (!stop_threads.load()) {
+      switch (type) {
+        case Iteration_Type::TOXAV_AUDIO:
+          toxav_audio_iterate(toxAV_.get());
+          std::this_thread::sleep_for(
+              std::chrono::milliseconds(toxav_audio_iteration_interval(toxAV_.get())));
+          break;
+
+        case Iteration_Type::TOXAV_VIDEO:
+          toxav_video_iterate(toxAV_.get());
+          std::this_thread::sleep_for(
+              std::chrono::milliseconds(toxav_video_iteration_interval(toxAV_.get())));
+          break;
+
+        case Iteration_Type::TOXAV_BOTH:
+          toxav_iterate(toxAV_.get());
+          std::this_thread::sleep_for(
+              std::chrono::milliseconds(toxav_iteration_interval(toxAV_.get())));
+          break;
+      }
+    }
+  }
+
+  std::thread tox_thread;
+  std::thread audio_thread;
+  std::thread video_thread;
+  std::thread av_thread;
+
+  std::mutex tox_loop_lock;
+
+  Tox_Ptr tox_;
+  bool combined_;
+  ToxAV_Ptr toxAV_;
+
+  std::atomic_bool stop_threads;
+  std::atomic_bool incomming;
+  std::atomic_uint32_t call_state;
+
+  std::atomic<TimePoint> call_start;
+  std::atomic_bool in_call;
+
+  std::atomic_bool video_received;
+  std::atomic_bool audio_received;
+  std::string name_;
+};
+
+struct DUMMY_PCM {
+  static constexpr size_t sample_count = 960;
+  static constexpr int16_t pcm[sample_count] = {0};
+  static constexpr uint8_t channels = 1;
+  static constexpr uint32_t sampling_rate = 48000;
+};
+
+struct DUMMY_VIDEO {
+  // https://en.wikipedia.org/wiki/Graphics_display_resolution#QQVGA size
+  static constexpr uint16_t width = 160;
+  static constexpr uint16_t height = 120;
+
+  static constexpr uint8_t y[width * height] = {0};
+  static constexpr uint8_t u[width / 2 * height / 2] = {0};
+  static constexpr uint8_t v[width / 2 * height / 2] = {0};
+};
+
+// FIXME: once we upgrade to C++17, remove these, reason: https://stackoverflow.com/a/28846608
+constexpr std::chrono::duration<double> AV_State::AUTO_HANGUP_TIME;
+constexpr int16_t DUMMY_PCM::pcm[];
+constexpr uint8_t DUMMY_VIDEO::y[];
+constexpr uint8_t DUMMY_VIDEO::u[];
+constexpr uint8_t DUMMY_VIDEO::v[];
+
+}  // namespace
+
+static void test_av(bool combined) {
+  std::cout << "Testing Audio and Video in" << (combined ? "combined" : "separate") << " threads"
+            << std::endl;
+  auto toxes = prepareNetwork(2);
+
+  AV_State alice(std::move(toxes[0]), "alice", false);
+  AV_State bob(std::move(toxes[1]), "bob", false);
+
+  // Let alice call bob
+  {
+    std::lock_guard<std::mutex>(alice.getToxLoopLock());
+    Toxav_Err_Call err;
+    ck_assert(
+        toxav_call(alice.getToxAV(), 0, AV_State::TEST_A_BITRATE, AV_State::TEST_V_BITRATE, &err));
+    ck_assert(err == TOXAV_ERR_CALL_OK);
+  }
+
+  std::cout << "alice started a call" << std::endl;
+
+  auto poll_state = [](AV_State &av, uint32_t expected, uint32_t max_tries,
+                       uint32_t delay_ms) -> bool {
+    uint32_t i;
```

Comment:
Unfortunately, this is C++, and cimple doesn't understand C++, so here comes the manual diagnostic:

```
toxav_mt_test.cc:408: variable `i' can be reduced in scope
toxav_mt_test.cc:410:   possibly to here
```